--- usbasp.c	Thu Sep 15 07:37:05 2011
+++ usbasp.c	Sat Apr  7 19:34:33 2012
@@ -579,6 +579,7 @@
 {
   unsigned char cmd[4];
   unsigned char res[4];
+  int i;
 
   if (p->op[AVR_OP_CHIP_ERASE] == NULL) {
     fprintf(stderr, "chip erase instruction not defined for part \"%s\"\n",
@@ -590,7 +591,8 @@
 
   avr_set_bits(p->op[AVR_OP_CHIP_ERASE], cmd);
   pgm->cmd(pgm, cmd, res);
-  usleep(p->chip_erase_delay);
+  // delay long enough for USBaspLoader-tiny85 to erase all pages individually
+  usleep(500000);
   pgm->initialize(pgm, p);
 
   return 0;
@@ -688,6 +690,10 @@
   } else {
      blocksize = USBASP_WRITEBLOCKSIZE;
   }
+
+  /* set blocksize to pagesize, so there can be a pause in USB traffic between pages to allow USBaspLoader-tiny85 to
+   * write the page to flash before needing to respond to the next USB message */
+  blocksize = page_size;
 
   while (wbytes) {
 
